app/screens/create_or_edit_channel/create_or_edit_channel.tsx

```diff
@@ -0,0 +1,254 @@
+
+type Props = {
+    componentId: string;
+    channel?: ChannelModel;
+    channelInfo?: ChannelInfoModel;
+}
+
+const CLOSE_BUTTON_ID = 'close-channel';
+const EDIT_BUTTON_ID = 'update-channel';
+const CREATE_BUTTON_ID = 'create-channel';
+
+enum RequestActions {
+    START = 'Start',
+    COMPLETE = 'Complete',
+    FAILURE = 'Failure',
+}
+
+interface RequestState {
+    error: string;
+    saving: boolean;
+}
+
+interface RequestAction {
+    type: RequestActions;
+    error?: string;
+}
+
+const close = (componentId: string): void => {
+    Keyboard.dismiss();
+    dismissModal({componentId});
+};
+
+const isDirect = (channel?: ChannelModel): boolean => {
+    return channel?.type === General.DM_CHANNEL || channel?.type === General.GM_CHANNEL;
+};
+
+export const makeCloseButton = (icon: any) => {
+    return buildNavigationButton(CLOSE_BUTTON_ID, 'close.more_direct_messages.button', icon);
+};
+
+const CreateOrEditChannel = ({
+    componentId,
+    channel,
+    channelInfo,
+}: Props) => {
+    const intl = useIntl();
+    const {formatMessage} = intl;
+    const theme = useTheme();
+    const serverUrl = useServerUrl();
+
+    const editing = Boolean(channel);
+
+    const [type, setType] = useState<ChannelType>(channel?.type as ChannelType || General.OPEN_CHANNEL);
+    const [canSave, setCanSave] = useState(false);
+
+    const [displayName, setDisplayName] = useState<string>(channel?.displayName || '');
+    const [purpose, setPurpose] = useState<string>(channelInfo?.purpose || '');
+    const [header, setHeader] = useState<string>(channelInfo?.header || '');
+
+    const [appState, dispatch] = useReducer((state: RequestState, action: RequestAction) => {
+        switch (action.type) {
+            case RequestActions.START:
+                return {
+                    error: '',
+                    saving: true,
+                };
+            case RequestActions.COMPLETE:
+                return {
+                    error: '',
+                    saving: false,
+                };
+            case RequestActions.FAILURE:
+                return {
+                    error: action.error,
+                    saving: false,
+                };
+
+            default:
+                return state;
+        }
+    }, {
+        error: '',
+        saving: false,
+    });
+
+    const rightButton = useMemo(() => {
+        const base = buildNavigationButton(
+            editing ? EDIT_BUTTON_ID : CREATE_BUTTON_ID,
+            'edit_channel.save.button',
+            undefined,
+            editing ? formatMessage({id: 'mobile.edit_channel', defaultMessage: 'Save'}) : formatMessage({id: 'mobile.create_channel', defaultMessage: 'Create'}),
+        );
+        base.enabled = canSave;
+        base.showAsAction = 'always';
+        base.color = theme.sidebarHeaderTextColor;
+        return base;
+    }, [editing, theme.sidebarHeaderTextColor, intl.locale, canSave]);
```

Comment:
```suggestion
    }, [editing, theme.sidebarHeaderTextColor, intl, canSave]);
```
..since it would also depends on `formatMessage`

Reply:
In theory, formatMessage would only change if we are changing the locale. Nevertheless, yes, let's play it safe and use `intl`.